### PR TITLE
fix syntax highlighting for `module _ = ` and add syntax tests

### PR DESCRIPTION
**Pull request by copy**
cc @Maelan

### CODE REVIEW

File: syntax-testcases/first_class_module.ml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module M1 : sig end = (struct let x = 0 end : sig end)
+(* => last signature: `sig` is linted as a type constructor, end `end` as an error *)
```
Comment (Maelan) [inline syntax-testcases/first_class_module.ml:2]: typo: ``and `end` ``

Comment (Maelan) [reply syntax-testcases/first_class_module.ml:2]: This test is redundant with `syntax-testcases/module_paren1.ml`

File: syntax-testcases/functor1.ml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module M : sig
+  module type F
+    = functor () (X : Ord) -> sig end (* not ok *)
```
Comment (Maelan) [inline syntax-testcases/functor1.ml:3]: This comment and the other ones in this file refer to linting as it was done *before* PR #95. Now all these tests work.

**Review by Maelan — COMMENTED** [review]
Thank you @copy, starting to add tests is a great idea indeed. :-)

**Comment by Maelan** [comment]
Perhaps leave a reference to relevant GitHub issue/pull numbers in comments?

File: syntax-testcases/functor1.ml (hunk 2 above)
Comment (copy) [reply syntax-testcases/functor1.ml:3]: Some of the equal signs seem to not be highlighted, but I guess that's a minor issue.

**Comment by copy** [comment]
> Perhaps leave a reference to relevant GitHub issue/pull numbers in comments?

Good idea, I've linked the broken tests to open issues.

**Comment by copy** [comment]
Merging, as tests should be fine and the fix is trivial.

**Pull request merged**